pkg/lightning/tikv/local_sst_writer_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/cockroachdb/pebble/sstable"
+	"github.com/cockroachdb/pebble/vfs"
+	"github.com/google/uuid"
+	rockssst "github.com/lance6716/pebble/sstable"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	sortedKVs = [][2][]byte{
+		{[]byte("a"), []byte("1")},
+	}
+	ts uint64 = 1
+)
+
+func TestGRPCWriteToTiKV(t *testing.T) {
+	t.Skip(`This is a manual test. You can use tiup playground and run this test.
+After the test is finished, find the SST files in the import directory of the TiKV node.`)
+
+	ctx := context.Background()
+	pdAddrs := []string{"127.0.0.1:2379"}
+
+	metas, err := write2ImportService4Test(ctx, pdAddrs, sortedKVs, ts)
+	require.NoError(t, err)
+	for _, meta := range metas {
+		t.Logf("meta UUID: %v", uuid.UUID(meta.Uuid).String())
+	}
+}
+
+func TestPebbleWriteSST(t *testing.T) {
+	//t.Skip("this is a manual test")
+
+	sstPath := "/tmp/test-write.sst"
+	writer, err := newWriteCFWriter(sstPath, ts, &rockssst.Identity{
+		DB:                 "SST Writer",
+		Host:               "lance6716-nuc10i7fnh",
+		Session:            "DS38NDUWK5HLG8SSL5M7",
+		OriginalFileNumber: 1,
+	})
+	require.NoError(t, err)
+
+	for _, kv := range sortedKVs {
+		err = writer.set(kv[0], kv[1])
+		require.NoError(t, err)
+	}
+
+	err = writer.close()
+	require.NoError(t, err)
+
+	f, err := vfs.Default.Open(sstPath)
+	require.NoError(t, err)
+	readable, err := rockssst.NewSimpleReadable(f)
+	require.NoError(t, err)
+	reader, err := rockssst.NewReader(readable, rockssst.ReaderOptions{})
+	require.NoError(t, err)
+	defer reader.Close()
+
+	layout, err := reader.Layout()
+	require.NoError(t, err)
+
+	infos := layout.BlockInfos(reader)
+	expected := `
+[
+	{"Offset":0,"Length":42,"Name":"data","Compression":0,"Checksum":2258416982},
```

Comment:
Can we calculate the checksum on the fly instead of a hard-coded value?
(Just curious, I don't know if it's necessary.)

Reply:
> a standard, TiKV-generated value

I mean,  can me calculate this value or get this value from somewhere? 🤔

Reply:
I don't understand how you suppose to calculate it

Reply:
Never mind, I must have misunderstood something Orz.

Reply:
I mean we should have 2 things to compare. This unit test compares the "generated file result" with "RocksDB's standard file result". The former one is already a computed value which is generated from given KV pairs and new SST writer. 

RocksDB's result is manually retrieve. Maybe you mean we can automatically retrieve that value? It's very complex, including starting a tiup playground, running `TestGRPCWriteToTiKV`, find the SST file in data-dir of TiKV and use `TestDebugReadSST` to get the result. I'll leave it for future when we need a CI to ensure no behaviour change at TiKV side.